test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1588 @@
+	cgroupMemLimit := Cgroupv2MemLimit
+	cgroupCPULimit := Cgroupv2CPULimit
+	cgroupCPURequest := Cgroupv2CPURequest
+	if !podOnCgroupv2Node {
+		cgroupMemLimit = CgroupMemLimit
+		cgroupCPULimit = CgroupCPUQuota
+		cgroupCPURequest = CgroupCPUShares
+	}
+	verifyCgroupValue := func(cName, cgPath, expectedCgValue string) bool {
+		ginkgo.By("Sleep " + PollInterval.String() + " seconds")
+		time.Sleep(PollInterval)
+		mycmd := fmt.Sprintf("head -n 1 %s", cgPath)
+		cgValue, _, err := ExecCommandInContainerWithFullOutput(ctx, f, pod.Name, cName, "/bin/sh", "-c", mycmd)
+		framework.Logf("Namespace %s Pod %s Container %s - looking for cgroup value %s in path %s",
+			pod.Namespace, pod.Name, cName, expectedCgValue, cgPath)
+		if flagError {
+			framework.ExpectNoError(err, "failed to find expected value '%s' in container cgroup '%s'",
+				expectedCgValue, cgPath)
+		}
+		cgValue = strings.Trim(cgValue, "\n")
+		if flagError {
+			gomega.Expect(cgValue).Should(gomega.Equal(expectedCgValue), "cgroup value")
+		}
+		if cgValue != expectedCgValue {
+			return false
+		}
+		return true
+	}
+	for _, ci := range tcInfo {
+		if ci.Resources == nil {
+			continue
+		}
+		tc, _ := makeTestContainer(ci)
+		if tc.Resources.Limits != nil || tc.Resources.Requests != nil {
+			var cpuShares int64
+			var cpuLimitString, memLimitString string
+			memLimitInBytes := tc.Resources.Limits.Memory().Value()
+			cpuRequest := tc.Resources.Requests.Cpu()
+			cpuLimit := tc.Resources.Limits.Cpu()
+			if cpuRequest.IsZero() && !cpuLimit.IsZero() {
+				cpuShares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
+			} else {
+				cpuShares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
+			}
+			cpuQuota := kubecm.MilliCPUToQuota(cpuLimit.MilliValue(), kubecm.QuotaPeriod)
+			if cpuLimit.IsZero() {
+				cpuQuota = -1
+			}
+			cpuLimitString = strconv.FormatInt(cpuQuota, 10)
+			if podOnCgroupv2Node {
+				if cpuLimitString == "-1" {
+					cpuLimitString = "max"
+				}
+				cpuLimitString = fmt.Sprintf("%s %s", cpuLimitString, CPUPeriod)
+			}
+			memLimitString = strconv.FormatInt(memLimitInBytes, 10)
+			if podOnCgroupv2Node && memLimitString == "0" {
+				memLimitString = "max"
+			}
+			if memLimitString != "0" {
+				if !verifyCgroupValue(ci.Name, cgroupMemLimit, memLimitString) {
+					return false
+				}
+			}
+			if !verifyCgroupValue(ci.Name, cgroupCPULimit, cpuLimitString) {
+				return false
+			}
+			if podOnCgroupv2Node {
+				// convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight value
+				cpuShares = int64(1 + ((cpuShares-2)*9999)/262142)
+			}
+			if !verifyCgroupValue(ci.Name, cgroupCPURequest, strconv.FormatInt(cpuShares, 10)) {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func waitForPodResizeActuation(ctx context.Context, f *framework.Framework, c clientset.Interface, podClient *e2epod.PodClient, pod, patchedPod *v1.Pod, expectedContainers []TestContainerInfo) *v1.Pod {
+
+	waitForContainerRestart := func() error {
```

Comment:
Use [pods.WaitForPodCondition](https://github.com/kubernetes/kubernetes/blob/a5753f60d679ea9d01263cca2858f9ceb75ae1fa/test/e2e/framework/pod/wait.go#L290) in place of all these polling funcitons.

Reply:
Alternatively, write a matcher and use gomega.Eventually

Reply:
Done